### ISSUE TITLE
Issue with File Upload Size Limit in stlite

### ISSUE THREAD

**Issue by elloza** [whitphx/stlite#1203]
Hello,

I am trying to upload a file larger than 3GB using a component in stlite. While this works perfectly in Streamlit, attempting the same in stlite results in the following error:

```
NotReadableError: The requested file could not be read, typically due to permission problems that have occurred after a reference to a file was acquired
```

I am unsure if this is a limitation of stlite or an error on my part. 

To handle large files, I added the following configuration in the `stlite` mount:

```javascript
streamlitConfig: {
    "server.maxUploadSize": "5000",
},
```

Thank you for your help, and congratulations on such an amazing project!

**Comment by whitphx** [comment]
Hi, it may be that the browser limits the memory or storage capacity outside Stlite,
while I haven't investigated it deeply yet.

**Comment by whitphx** [comment]
Can you provide the full error log and the stack trace around the error?

**Comment by elloza** [comment]
You are right, the memory limit of 2GB seems to be the main limitation (https://github.com/pyodide/pyodide/discussions/3796).

Sorry for the delay in responding and thanks for your time.
